File: UnitTesting/SpecFlow-Examples-master/BowlingKata/BowlingKata-FluentAssertions/Bowling.SpecFlow/StepDefinitions/BowlingSteps.cs
using Bowling.Specflow.Drivers;
using TechTalk.SpecFlow;

namespace Bowling.Specflow.StepDefinitions
{
    [Binding]
    public class BowlingSteps
    {
        private readonly BowlingDriver _driver;

        public BowlingSteps(BowlingDriver driver)
        {
            _driver = driver;
        }

        [Given(@"a new bowling game")]
        public void GivenANewBowlingGame()
        {
            _driver.NewGame();
        }

        [When(@"all of my balls are landing in the gutter")]
        public void WhenAllOfMyBallsAreLandingInTheGutter()
        {
            _driver.Roll(0, 20);
        }

        [When(@"all of my rolls are strikes")]
        public void WhenAllOfMyRollsAreStrikes()
        {
            _driver.Roll(10, 12);
        }

        [Then(@"my total score should be (\d+)")]
        public void ThenMyTotalScoreShouldBe(int score)
        {
            _driver.CheckScore(score);
        }

        [When(@"I roll (\d+)")]
        public void WhenIRoll(int pins)
        {
            _driver.Roll(pins, 1);
        }

        [When(@"I roll (\d+) and (\d+)")]
        public void WhenIRoll(int pins1, int pins2)
        {
            _driver.Roll(pins1, pins2, 1);
        }

        [When(@"I roll (\d+) times (\d+) and (\d+)")]
        public void WhenIRollSeveralTimes2(int rollCount, int pins1, int pins2)
        {
            _driver.Roll(pins1, pins2, rollCount);
        }

        [When(@"I roll the following series:(.*)")]
        public void WhenIRollTheFollowingSeries(string series)
        {
            _driver.RollSeries(series);
        }

        [When(@"I roll")]
        public void WhenIRoll(Table rolls)
        {
            _driver.RollSeries(rolls);
        }
    }
}

File: UnitTesting/SpecFlow-Examples-master/BowlingKata/BowlingKata-VB-MsTest/Bowling.Specflow/StepDefinitions/BowlingSteps.vb
Imports Bowling.Bowling
Imports Bowling.Specflow.Drivers
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports TechTalk.SpecFlow

Namespace StepDefinitions
    <Binding>
    Public Class BowlingSteps
        Private ReadOnly _driver As BowlingDriver

        Public Sub New(driver As BowlingDriver)
            _driver = driver
        End Sub

        <Given("a new bowling game")>
        Public Sub GivenANewBowlingGame()
            _driver.NewGame()
        End Sub

        <[When]("all of my balls are landing in the gutter")>
        Public Sub WhenAllOfMyBallsAreLandingInTheGutter()
            _driver.Roll(0, 20)
        End Sub

        <[When]("all of my rolls are strikes")>
        Public Sub WhenAllOfMyRollsAreStrikes()
            _driver.Roll(10, 12)
        End Sub

        <[Then]("my total score should be (\d+)")>
        Public Sub ThenMyTotalScoreShouldBe(score As Integer)
            _driver.CheckScore(score)
        End Sub

        <[When]("I roll (\d+)")>
        Public Sub WhenIRoll(pins As Integer)
            _driver.Roll(pins, 1)
        End Sub

        <[When]("I roll (\d+) and (\d+)")>
        Public Sub WhenIRoll(pins1 As Integer, pins2 As Integer)
            _driver.Roll(pins1, pins2, 1)
        End Sub

        <[When]("I roll (\d+) times (\d+) and (\d+)")>
        Public Sub WhenIRollSeveralTimes2(rollCount As Integer, pins1 As Integer, pins2 As Integer)
            _driver.Roll(pins1, pins2, rollCount)
        End Sub

        <[When]("I roll the following series:(.*)")>
        Public Sub WhenIRollTheFollowingSeries(series As String)
            _driver.RollSeries(series)
        End Sub

        <[When]("I roll")>
        Public Sub WhenIRoll(rolls As Table)
            _driver.RollSeries(rolls)
        End Sub
    End Class
End Namespace


